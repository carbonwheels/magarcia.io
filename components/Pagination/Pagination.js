import Link from 'next/link';
import PropTypes from 'prop-types';
import { range } from '../../lib/utils';
import css from './Pagination.module.css';

const Pagination = ({ total, page }) => (
  <div>
    {range(total).map(val => {
      if (page === val) {
        return (
          <a key={val} className={`${css.page} ${css.active}`}>
            {val}
          </a>
        );
      }
      return (
        <Link
          key={val}
          as={val === 1 ? '/' : `/blog/page/${val}`}
          href={val === 1 ? '/' : `/?page=${val}`}
        >
          <a className={css.page}>{val}</a>
        </Link>
      );
    })}
  </div>
);

Pagination.propTypes = {
  total: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired
};

export default Pagination;
