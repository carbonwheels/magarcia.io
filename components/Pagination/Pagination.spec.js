import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';

it('renders correctly', () => {
  const tree = renderer.create(<Pagination total={4} page={2} />).toJSON();
  expect(tree).toMatchSnapshot();
});
