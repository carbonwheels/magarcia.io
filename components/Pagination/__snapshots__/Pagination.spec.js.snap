// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <a
    className="page"
    href="/"
    onClick={[Function]}
  >
    1
  </a>
  <a
    className="page active"
  >
    2
  </a>
  <a
    className="page"
    href="/blog/page/3"
    onClick={[Function]}
  >
    3
  </a>
  <a
    className="page"
    href="/blog/page/4"
    onClick={[Function]}
  >
    4
  </a>
</div>
`;
